synadm/cli/user.py

```diff
@@ -554,3 +554,24 @@ def user_login_cmd(helper, user_id, expire_days, expire, expire_ts,
             raise SystemExit(1)
         else:
             helper.output(user_login)
+
+@user.command(name="shadow-ban")
+@click.argument("user_id", type=str)
+@click.option(
+    "-u", "--unban", is_flag=True, default=False, show_default=False,
+    help="Unban the specified user"
+)
+@click.pass_obj
+def user_shadow_ban(helper, user_id, unban):
+    """Shadow-ban or unban a user.
+
+    Useful for moderating malicious or egregiously abusive users.
+
+    A shadow-banned user will not receive any notification, but still will be
+    unable to contact anyone on the server. Use this as a tool of last resort since it
+    may lead to confusing or broken behaviour for the client. 
+    
+    Generally, it is more appropriate to ban or kick abusive users.
+    """
+    user_ban = helper.api.user_shadow_ban(user_id, unban)
+    helper.output(user_ban)
```

Comment:
This API seems to always return empty dict {} when no error occured. If you want but not strictly required you could prettify that for human mode. Something like this: https://github.com/Ascurius/synadm/blob/master/synadm/cli/user.py#L127-L134